packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -144,6 +148,22 @@ const isWithinAvailableHours = (
       return true;
     }
   }
+
+  // check if it is a date override
+  for (const dateOverride of dateOverrides) {
```

Comment:
we only checked if the slot is within working hours but it can also be a date override that is outside of the working hours